app/controllers/exercises_controller.rb

```diff
@@ -0,0 +1,77 @@
+class ExercisesController < ApplicationController
+  before_action :logged_in_user
+  before_action :load_courses, only: %i(new create)
+  before_action :load_exercise, :correct_answers, only: %i(show edit update)
+
+  def index
+    @exercises = Exercise.last_exercises.page(params[:page])
+      .per Settings.exercises.paging.paging_number
+  end
+  
+  def new
+    @exercise = Exercise.new
+    respond_to do |format|
+      format.html
+      format.js
+    end
+  end
+
+  def create
+    params[:exercise][:user_id] = current_user.id
+    course_id = params[:exercise][:course_id].to_i
+    
+    ActiveRecord::Base.transaction do 
```

Comment:
cái này hoạt động như thế nào

Reply:
cái này dùng để tránh dữ liệu rác, cái này hoạt động toàn vẹn dữ liệu hay nói cách khác là dảm bảo các câu sql modify DB đều thành công nếu chỉ 1 thao tác thất bại sẽ không thực hiên thao tác khác

Reply:
trả lời có vẻ hợp lý nhưng chưa thuyết phục lắm :v "dữ liệu được đưa vào cần đúng và đủ"